web/packages/teleport/src/Users/state.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Teleport
+ * Copyright (C) 2025 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  useEffect,
+  useRef,
+  useState,
+  type Dispatch,
+  type SetStateAction,
+} from 'react';
+import { useHistory } from 'react-router';
+
+export interface UsersUrlState {
+  search: string;
+  page: number;
+  username: string | null;
+}
+
+export function searchParamsToState(params: URLSearchParams): UsersUrlState {
+  const state: UsersUrlState = {
+    search: '',
+    page: 1,
+    username: null,
+  };
+
+  const search = params.get('search');
+  if (search) {
+    state.search = search;
+  }
+
+  const page = params.get('page');
+  if (page) {
+    const pageNum = parseInt(page, 10);
+    if (!isNaN(pageNum) && pageNum > 0) {
+      state.page = pageNum;
+    }
+  }
+
+  const username = params.get('username');
+  if (username) {
+    state.username = username;
+  }
+
+  return state;
```

Comment:
Consider simplifying this:

```suggestion
  const state: UsersUrlState = {
    search: params.get('search') ?? '',
    username: params.get('username'),
    page: 1,
  }
  
  const page = params.get('page');
  if (page) {
    const pageNum = parseInt(page, 10);
    if (!isNaN(pageNum) && pageNum > 0) {
      state.page = pageNum;
    }
  }

    return state;
```

Also, are pages counted starting from 1 or 0?